spec/factories/firm.rb

```diff
@@ -35,6 +35,19 @@
       in_person_advice_methods []
     end
 
+    factory :registered_firm do
+      parent factory: Firm
+    end
+
+    factory :onboarded_firm do
+      parent factory: :registered_firm
+    end
+
+    factory :not_onboarded_firm do
+      parent factory: :firm
+      email_address nil
+    end
+
```

Comment:
Ok, this feels better. It stops the spread of implementation details. I think this factory is due a rethink at some point though.